roles/confluent.kafka_broker/tasks/main.yml

```diff
@@ -109,18 +109,21 @@
     mode: 0640
     owner: "{{kafka_broker_user}}"
     group: "{{kafka_broker_group}}"
-  when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms"
+  when: ("'GSSAPI' in kafka_broker_sasl_enabled_mechanisms") or
+        (zookeeper_sasl_digestmd5_enabled)
   notify:
     - Restart Kafka
 
 - name: Create SCRAM Users
```

Comment:
This task runs on zookeeper[0] host, but uses the kafka_broker jaas file. Should it be using the zookeeper jaas file?
Are you testing on where zookeeper and kafka are on the same host?

Reply:
I think this one must be runned from kafka_broker[0].

But, look at this, even in original task it uses "kafka-configs --zookeeper **localhost:**" wich assumes that kafka and zookeeper are placed on the same host.